platform/darwin/src/MGLPolygon.mm

```diff
@@ -54,6 +55,17 @@ - (NSUInteger)hash {
     return [super hash] + [[self geoJSONDictionary] hash];
 }
 
+- (CLLocationCoordinate2D)coordinate {
+    CLLocationCoordinate2D centroid;
+    
+    // pole of inaccessibility
+    auto poi = mapbox::polylabel([self polygon]);
+    centroid.latitude = poi.y;
+    centroid.longitude = poi.x;
```

Comment:
In general, we should initialize simple structs like this using factory methods. This is especially important for geometry structs where it’s easy to confuse latitude and longitude or x and y. Create an `MGLLocationCoordinate2DFromPoint()` method similar to [`MGLPointFromLocationCoordinate2D()`](https://github.com/mapbox/mapbox-gl-native/blob/4b84ae119dd9489b6fbd6a11eca2102e0a79747b/platform/darwin/src/MGLGeometry_Private.h#L17) and call it here.